app/src/main/cpp/skyline/input/npad_device.cpp

```diff
@@ -188,8 +200,8 @@ namespace skyline::input {
 
         info.header.timestamp = util::GetTimeTicks();
         info.header.entryCount = std::min(static_cast<u8>(info.header.entryCount + 1), constant::HidEntryCount);
-        info.header.maxEntry = info.header.entryCount;
-        info.header.currentEntry = (info.header.currentEntry != constant::HidEntryCount - 1) ? info.header.currentEntry + 1 : 0;
+        info.header.maxEntry = info.header.entryCount - 1;
```

Comment:
is this needed for touch too?

Reply:
Yes. All HID devices have the same behavior